Gah with the last change I get what looks like being an unrelated failure:
```
FAIL: testDoAction (__main__.TestQgsActionManager)

test running action

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/qgis/QGIS/build/output/python/qgis/testing/__init__.py", line 219, in wrapper

    func(*args, **kwargs)

  File "/home/travis/build/qgis/QGIS/tests/src/python/test_qgsactionmanager.py", line 208, in testDoAction

    self.assertEqual(self.check_action_result(temp_file), 'test output')

AssertionError: '' != 'test output'

+ test output
```

In https://travis-ci.org/qgis/QGIS/jobs/240025470#L1209-L1221

--- a/python/plugins/db_manager/db_plugins/connector.py
+++ b/python/plugins/db_manager/db_plugins/connector.py
@@ -196,7 +196,7 @@ class DBConnector:
     @classmethod
     def quoteString(self, txt):
         """ make the string safe - replace ' with '' """
-        if hasattr(txt, '__iter__'):
+        if hasattr(txt, '__iter__') and not isinstance(txt, str):
             txts = list()
             for i in txt:
                 if i is None:

--- a/python/plugins/db_manager/db_plugins/connector.py
+++ b/python/plugins/db_manager/db_plugins/connector.py
@@ -211,7 +211,9 @@ class DBConnector:
     def getSchemaTableName(self, table):
         if not hasattr(table, '__iter__'):
             return (None, table)
-        elif len(table) < 2:
+        if isinstance(table, str):
+            table = table.split('.')
+        if len(table) < 2:
             return (None, table[0])
         else:
             return (table[0], table[1])

--- a/python/plugins/db_manager/db_plugins/postgis/connector.py
+++ b/python/plugins/db_manager/db_plugins/postgis/connector.py
@@ -60,7 +60,7 @@ class PostGisDBConnector(DBConnector):
 
         expandedConnInfo = self._connectionInfo()
         try:
-            self.connection = psycopg2.connect(expandedConnInfo.encode('utf-8'))
+            self.connection = psycopg2.connect(expandedConnInfo)
         except self.connection_error_types() as e:
             err = unicode(e)
             uri = self.uri()
@@ -79,7 +79,7 @@ class PostGisDBConnector(DBConnector):
 
                 newExpandedConnInfo = uri.connectionInfo(True)
                 try:
-                    self.connection = psycopg2.connect(newExpandedConnInfo.encode('utf-8'))
+                    self.connection = psycopg2.connect(newExpandedConnInfo)
                     QgsCredentials.instance().put(conninfo, username, password)
                 except self.connection_error_types() as e:
                     if i == 2:

